### PR TITLE
[CONTINT-4335] Add ECS agent version to agent flare

### PR DESCRIPTION
**Pull request by ewoodthomas**
### What does this PR do?

https://datadoghq.atlassian.net/browse/CONTINT-4335

Adds the `ECS Metadata` to the agent flare in a new file `ecs_metadata.json`. If not running on ECS, this file is not created. The only information present is the `ECS Agent Version` and `ECS Cluster Name`.

### Motivation

The `ECS Agent Version` would be useful for debugging support tickets.

### Additional Notes

Not sure if this change should have a release note. Primarily something that would be used by support and not customers, but let me know if one should be added in.

### Describe how to test/QA your changes

- Start up an ECS cluster and deploy the agent (image must be amd64 architecture)
- Check the `ECS Agent Version`
  - `aws ecs describe-container-instances --cluster <cluster name> --container-instances <cluster instance>`
- Exec into the agent task
  - `aws ecs execute-command \
  --cluster <cluster name> \
  --task <task id> \
  --container datadog-agent \
  --interactive \
  --command "/bin/bash"`
- Call `agent flare`
- Unzip the flare and check that `ecs_metadata.json` is created and verify the ECS agent version is present and the same as what was previously found

Additional QA:

- Deploy the agent on a kind cluster
- Exec into the agent and call `agent flare`
- Unzip the flare and verify that `ecs_metadata.json` is not created

### CODE REVIEW

**Review by hush-hush — CHANGES_REQUESTED** [review]
HI,

We should not modify the host metadata providers as it's critical to the backend and how we register new host.

If you want to extend the flare you should use a flare [provider](https://datadoghq.dev/datadog-agent/components/shared_features/flares/) instead of modifying the metadata.
The metadata are linked to product behavior in the backend and we add them to the flare for troubleshooting reasons.

File: pkg/flare/archive.go [hunk 1]
```diff
@@ -394,6 +396,23 @@ func getHealth() ([]byte, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	jsonValue, err := json.MarshalIndent(ecsMeta, "", "\t")
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonValue, nil
```
Comment (hush-hush) [inline pkg/flare/archive.go:413]:
```suggestion
	return json.MarshalIndent(ecsMeta, "", "\t")
```

**Review by hush-hush — APPROVED** [review]
Added a nit-pick but looks good to merge after that.

File: pkg/util/ecs/ecs.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	ecsMeta := MetaECS{
+		ECSCluster:      cluster,
+		ECSAgentVersion: version,
+	}
+	return &ecsMeta, nil
+}
+
+// GetECSInstanceMetadata fetches the ECS Instance metadata if running in ECS
+func GetECSInstanceMetadata(ctx context.Context) (string, string, error) {
```
Comment (sblumenthal) [inline pkg/util/ecs/ecs.go:41]: This seems like a redundant function in its current form, given that it is only being used by `NewECSMeta`. I feel that either:
1) The logic of this function can be moved to `NewECSMeta`, or
2) The function should be renamed to `getECSInstanceMetadata`, as right now there is no reason to expose this function

I am personally leaning towards the first one, but in either case the matching file signature should be removed from `no_ecs.go`

**Review by sblumenthal — APPROVED** [review]
One small note, otherwise it looks good

**Pull request merged**